plugins/modules/sonarr_quality_profile.py

```diff
@@ -230,58 +196,67 @@ def run_module():
                 result['id'] = 0
         module.exit_json(**result)
 
+    # Get schema
+    schema_client = sonarr.QualityProfileSchemaApi(module.api)
+    try:
+        want = schema_client.get_qualityprofile_schema()
+    except Exception as e:
+        module.fail_json('Error getting quality profile schema: %s' % to_native(e.reason), **result)
+
+    # Break apart quality groups and create a dict.
+    quality_groups_dict = OrderedDict()
+
+    for item in want.items:
+        if item.quality is None:
+            for quality in item.items:
+                quality.allowed = False
+                quality_groups_dict[quality.quality.name] = quality
+        else:
+            item.allowed = False
+            quality_groups_dict[item.quality.name] = item
```

Comment:
Haven't tried this out yet. However, if we start from this, will we still be able to create a custom group?

Reply:
Yep, that's done below. Take a look at the example. I just pull it apart at the start for easier processing and to have a consistent starting point with everything flat.